### ISSUE TITLE
General structure and API Questions

### ISSUE THREAD

**Issue by bumpingChris** [XHMM/react-native-cloud-store#32]
@XHMM @eveningkid 
Thank you @XHMM for maintaining this library! 
I'm new to iCloud syncing, and so have a few basic understanding questions. I hope you can help clarify them for me.
Is my understanding on the following correct?

1. On the mobile device, there are two locations where files are/can be stored:  
    1) Local path = app relative path (e.g. /var/mobile/Containers/Data/Application/1A6CCFDA-89DC-4FE7-8CF0-72180FD34FBB/Documents/my.db). 
    2) Cloud container path = directory where files are synced to the cloud. (e.g. /private/var/mobile/Library/Mobile Documents/iCloud~com~project~app/Documents/my.db). **Files here exist locally AND remotely?**

2. When we execute a method like CloudStore.download, we're getting the latest version of the file from the remote cloud container to the local cloud container? And, once downloaded, we need to use RNFS.copyFile to copy the file from the cloud container to a app-relative location (e.g. RNFS.copyFile(/private/var/mobile/Library/Mobile Documents/iCloud~com~project~app/Documents/my.db, /var/mobile/Containers/Data/Application/1A6CCFDA-89DC-4FE7-8CF0-72180FD34FBB/Documents/my.db)? Is it not good practice to reference files in the local cloud container directly from the app?

3. What's the difference between CloudStore.download and CloudStore.startDownloadingUbiquitousItem ?

Thank you in advance for your help in all this!

_Originally posted by @bumpingChris in https://github.com/XHMM/react-native-cloud-store/issues/27#issuecomment-2484580192_

**Comment by XHMM** [comment]
> 1. Files here exist locally AND remotely?

Yes when you downloaded it, due to it exists in the cloud folder, the file content maybe changed by apple system such as sync with new version from other apple devices.

> 2. When we execute a method like CloudStore.download, we're getting the latest version of the file from the remote cloud container to the local cloud container?    And, once downloaded, we need to use RNFS.copy...?   Is it not good practice to reference files in the local cloud container directly from the app?

Yes;
Yes;
No, becuase files in the cloud folder may changed as said in 1.

> 3. What's the difference between CloudStore.download and CloudStore.startDownloadingUbiquitousItem ?

`startDownloadingUbiquitousItem` is just a wrapper of swift-side api it's 'low-level', `download` is a 'high-level' api of it, you may just use `download`

**Comment by bumpingChris** [comment]
@XHMM Thank you so much for the clarifications.

**Comment by bumpingChris** [comment]
@XHMM Sorry to bother you. I have another general question about this architecture. 

I'm trying to figure out if readDir read all files from the REMOTE iCloud directory or only the LOCAL iCloud directory.   

Let's say:
1) In the REMOTE iCloud directory, device A added a file "XYZ" to it. 
2) In device B, does the function readDir or download get that "XYZ" file (as it doesn't exist in its local cloud directory). 

Thank you for your help.
